http/client/src/main/java/org/wildfly/security/http/client/ElytronHttpClient.java

```diff
@@ -0,0 +1,211 @@
+
+        String response;
+        if (qop == null) {
+            response = digestHttpMechanismUtil.computeDigestWithoutQop(path, nonce, userName, password, algorithm, realm, "GET");
+        } else {
+            response = digestHttpMechanismUtil.computeDigestWithQop(path, nonce, "0a4f113b", "00000001", userName, password, algorithm, realm, qop, "GET");
+        }
+
+        HttpRequest request2 = HttpRequest
+                .newBuilder()
+                .uri(new URI(uri))
+                .header(AUTHORIZATION, "Digest " +
+                        "username=\"" + userName + "\", " +
+                        "realm=\"" + realm + "\"," +
+                        "nonce=\"" + nonce + "\", " +
+                        "uri=\"" + path + "\", " +
+                        "qop=\"" + qop + "\", " +
+                        "nc=00000001, " +
+                        "cnonce=\"0a4f113b\", " +
+                        "response=\"" + response + "\", " +
+                        "opaque=\"" + opaque + "\", " +
+                        "algorithm=" + algorithm)
+                .build();
+        return request2;
+
+    }
+
+    public HttpRequest getRequest(String uri) throws Exception {
+        String username = httpMechClientConfigUtil.getUsername(new URI(uri));
+        String password = httpMechClientConfigUtil.getPassword(new URI(uri));
+        String AuthType = httpMechClientConfigUtil.getHttpAuthenticationType(new URI(uri));
+        String AuthHeader = null;
+        if (AuthType.equalsIgnoreCase("basic")) {
+            AuthHeader = basicAuth(username, password);
+        }
+        HttpRequest request = HttpRequest
+                .newBuilder()
+                .uri(new URI(uri))
+                .header(AUTHORIZATION, AuthHeader)
+                .build();
+
+        return request;
+    }
+
+    public HttpResponse evaluateBasicMechanism(String uri) throws Exception {
+        HttpRequest request = HttpRequest
+                .newBuilder()
+                .uri(new URI(uri))
+                .header(AUTHORIZATION, basicAuth(userName, password))
+                .build();
+        HttpResponse response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        return response;
+    }
+    public HttpResponse evaluateNoAuthMechanism(String uri) throws Exception{
+        HttpRequest request = HttpRequest
+                .newBuilder()
+                .uri(new URI(uri))
+                .build();
+
+        HttpResponse response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        return response;
+    }
+
+    public HttpResponse connect(String uri) throws Exception {
+        userName = httpMechClientConfigUtil.getUsername(new URI(uri));
+        password = httpMechClientConfigUtil.getPassword(new URI(uri));
+        HttpResponse response = null;
+
+        if (lastURI != null && lastURI.equals(uri)) {
+            switch (previousMechanism) {
```

Comment:
@keshav-725 Only digest mechanism has nonce counts. So we do not have to use the switch here ? it can just be smth like

```
if ( lastURI != null && lastURI.equals(uri) && previousMechanism.equals(Digest)) 
   {process the request with same `nonce` as before and higher `nc`}
```